en/debug/plotting_realtime_uorb_data.md

```diff
@@ -0,0 +1,118 @@
+# Plotting uORB Topic Data in Real Time using PlotJuggler
+
+This topic shows how you can graph the "live" values of [uORB topics](../msg_docs/index.md) (in real time) using [PlotJuggler](../log/flight_log_analysis.md#plotjuggler) and the _uXRCE-DDS Agent_.
+
+This technique uses PX4 [uXRCE-DDS](../middleware/uxrce_dds.md) middleware to export uORB topics as ROS2 topics, which can then be read and plotted by PlotJuggler as they change (PlotJuggler cannot directly read uORB topics, but the values of the corresponding ROS 2 topics are the same).
+
+The video below demonstrates this for a simulated vehicle — the approach works equally well on real hardware.
+
+<video src="../../assets/debug/realtime_debugging/realtime_debugging.mp4" width="720" controls></video>
+
+## Prerequisites
+
+Follow the [ROS 2 Installation & Setup](../ros2/user_guide.md#installation-setup) instructions in the _ROS2 user guide_ to install:
+
+- ROS 2
+- [Micro XRCE-DDS Agent](../ros2/user_guide.md#setup-micro-xrce-dds-agent-client)
+- [PX4/px4_msgs](https://github.com/PX4/px4_msgs): PX4/ROS2 shared message definitions.
+- PX4 source code and build the simulator.
+
+  ::: tip
+  If you're using real hardware instead of the simulator, you will only need PX4 source code if you need to change the set of topics that are published to ROS 2 (only a subset of uORB topics are published by default).
+  :::
+
+You will also need to install:
+
+- [PlotJuggler for ROS2](https://github.com/facontidavide/PlotJuggler)
+
+  ::: tip
+  Use the Debian packages (the snap files are not supported).
+  :::
+
+## Usage
+
+First we need to build a ROS 2 workspace that includes the `px4_msgs` that correspond to the PX4 build to be monitored, and then launch PlotJuggler from within that workspace.
+This allows ROS 2 and PlotJuggler to interpret the messages.
+If you're using unmodified PX4, the definitions from [PX4/px4_msgs](https://github.com/PX4/px4_msgs) can be used.
+
+::: info
+This is the same process as covered in [Build ROS 2 Workspace](../ros2/user_guide.md#build-ros-2-workspace) in _ROS 2 Installation & Setup_.
+:::
+
+Assuming your ROS 2 workspace is `~/ros2_ws/`, fetch and build the `px4_msgs` package in a terminal as shown:
+
+```sh
+cd ~/ros2_ws/src/
+git clone https://github.com/PX4/px4_msgs.git
+cd ..
```

Comment:
I added this, because the instructions in the user guide do NOT use `ros2_ws` as a workspace, and it is a lot for a newbie to remember how they access and use a workspace.

Reply:
yea, there i'm not sure if its good that we use `ws_sensor_combined` in the ROS2 HowTo, as the officials ros HowTo's always use `ros2_ws `

Reply:
Yeah. I recall doing this for the reason that it wasn't clear in any of the ROS docs whether ros2_ws was a magic location or  just a convention. I quite like having something in our docs that is different.